src/Http/Http.Extensions/src/HeaderDictionaryTypeExtensions.cs

```diff
@@ -132,45 +132,85 @@ internal static void SetDate(this IHeaderDictionary headers, string name, DateTi
+
+    private static ContentDispositionHeaderValue? ParseCacheContentDispositionHeaderValue(string value) => ContentDispositionHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static ContentRangeHeaderValue? ParseCacheContentRangeHeaderValue(string value) => ContentRangeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static MediaTypeHeaderValue? ParseCacheMediaTypeHeaderValue(string value) => MediaTypeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static RangeConditionHeaderValue? ParseCacheRangeConditionHeaderValue(string value) => RangeConditionHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static RangeHeaderValue? ParseCacheRangeHeaderValue(string value) => RangeHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static EntityTagHeaderValue? ParseCacheEntityTagHeaderValue(string value) => EntityTagHeaderValue.TryParse(value, out var result) ? result : null;
+
+    private static DateTimeOffset? ParseCacheDateTimeOffset(string value) => HeaderUtilities.TryParseDate(value, out var result) ? result : null;
+
+    private static long? ParseCacheInt64(string value) => HeaderUtilities.TryParseNonNegativeInt64(value, out var result) ? result : null;
+
+    private static IList<MediaTypeHeaderValue> ParseMediaTypeHeaderValue(IList<string> value) =>
+        MediaTypeHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<MediaTypeHeaderValue>();
+
+    private static IList<StringWithQualityHeaderValue> ParseStringWithQualityHeaderValue(IList<string> value) =>
+        StringWithQualityHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<StringWithQualityHeaderValue>();
+
+    private static IList<CookieHeaderValue> ParseCookieHeaderValue(IList<string> value) =>
+        CookieHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<CookieHeaderValue>();
+
+    private static IList<EntityTagHeaderValue> ParseEntityTagHeaderValue(IList<string> value) =>
+        EntityTagHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<EntityTagHeaderValue>();
+
+    private static IList<SetCookieHeaderValue> ParseSetCookieHeaderValue(IList<string> value) =>
+        SetCookieHeaderValue.TryParseList(value, out var result) ? result : Array.Empty<SetCookieHeaderValue>();
 
     internal static T? Get<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] T>(this IHeaderDictionary headers, string name)
     {
         ArgumentNullException.ThrowIfNull(headers);
-
         var value = headers[name];
 
         if (StringValues.IsNullOrEmpty(value))
         {
             return default(T);
         }
 
-        if (KnownParsers.TryGetValue(typeof(T), out var temp))
+        if (typeof(T) == typeof(CacheControlHeaderValue))
```

Comment:
Personally, I might extract a local for `value.ToString()`.

Reply:
I might also put a comment in front of the if explaining that it was structured this way for perf, so someone doesn't come along and "clean it up".

Reply:
Something like?
```suggestion
        // Comparison of typeof(T) is optimized by JIT.
        if (typeof(T) == typeof(CacheControlHeaderValue))
```

Reply:
I added the comments and hoisted the `ToString()` - that was very evident not sure how I missed that (...copilot suggestions).